src/app_modes/topic_managment.rs

```diff
@@ -0,0 +1,365 @@
+            |i| {[i.topic.clone(),  "visualization_msgs/MarkerArray".to_string()]}
+        ).collect();
+        let active_marker_topics: Vec<[String; 2]> = config.marker_topics.iter().map(
+            |i| {[i.topic.clone(),  "visualization_msgs/Marker".to_string()]}
+        ).collect();
+        let active_pose_stamped_topics: Vec<[String; 2]> = config.pose_stamped_topics.iter().map(
+            |i| {[i.topic.clone(),  "geometry_msgs/PoseStamped".to_string()]}
+        ).collect();
+        let active_pose_array_topics: Vec<[String; 2]> = config.pose_array_topics.iter().map(
+            |i| {[i.topic.clone(),  "geometry_msgs/PoseArray".to_string()]}
+        ).collect();
+        let active_path_topics: Vec<[String; 2]> = config.path_topics.iter().map(
+            |i| {[i.topic.clone(),  "nav_msgs/Path".to_string()]}
+        ).collect();
+        // Collect them into a big list
+        let all_active_topics = [
+            active_laser_topics,
+            active_marker_array_topics,
+            active_marker_topics,
+            active_pose_stamped_topics,
+            active_pose_array_topics,
+            active_path_topics
+        ].concat();
+
+        
+        // We could get this from config, but would need some breaking changes in config
+        let supported_topic_types = vec![
+            "sensor_msgs/LaserScan".to_string(),
+            "visualization_msgs/MarkerArray".to_string(),
+            "visualization_msgs/Marker".to_string(),
+            "geometry_msgs/PoseStamped".to_string(),
+            "geometry_msgs/PoseArray".to_string(),
+            "nav_msgs/Path".to_string(),
+        ];
+        // Collect all topics, which:
+        //  - are supported
+        //  - are inactive
+        let supported_topics: Vec<[String; 2]>= rosrust::topics()
+            .unwrap()
+            .iter()
+            .map(|topic|{
+                [topic.name.to_string(),
+                 topic.datatype.to_string()
+                ]
+            })
+            .filter(|el| supported_topic_types.contains(&el[1].to_string()))
+            .filter(|el| !all_active_topics.contains(&el))
+            .collect();
+
+
+        // Fill the state manager with active and supported topics
+        TopicManager {
+            availible_topics: SelectableTopics::new(supported_topics),
+            selected_topics: SelectableTopics::new(all_active_topics),
+            config: config,
+            selection_mode: true,
+        }
+    }
+
+    pub fn shift_active_element_right(&mut self) {
+        if self.availible_topics.items.is_empty() {return}
+        let x = self.availible_topics.pop();
+        self.selected_topics.add(x);
+    }
+    pub fn shift_active_element_left(&mut self) {
+        if self.selected_topics.items.is_empty() {return}
+        let x = self.selected_topics.pop();
+        self.availible_topics.add(x);
+    }
+
+    pub fn save(&mut self) {
```

Comment:
the user needs to get feedback if changes to the config were made and if they were stored

Reply:
Done!